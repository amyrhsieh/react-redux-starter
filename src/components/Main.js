import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Home from './Home';
import Events from './Events';
import Page2 from './Page2';

const Main = () => (
  <main className="container">
    <Switch>
      <Route exact path='/' component={Home}/>
      <Route path='/events' component={Events}/>
      <Route path='/page2' component={Page2}/>
    </Switch>
  </main>
);

export default Main;
