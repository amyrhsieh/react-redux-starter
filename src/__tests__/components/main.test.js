import React from 'react';
import Main from 'Components/Main.js';
import { shallow } from 'enzyme';

describe('Main (Snapshot)', () => {
  it('Main component renders Main', () => {
    const wrapper = shallow(<Main />);
    expect(wrapper).toMatchSnapshot();
  });
});
