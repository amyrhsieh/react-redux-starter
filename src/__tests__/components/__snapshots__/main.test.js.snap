// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main (Snapshot) Main component renders Main 1`] = `
<main
  className="container"
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Events)",
          "type": [Function],
        }
      }
      path="/events"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Blog)",
          "type": [Function],
        }
      }
      path="/blog"
    />
  </Switch>
</main>
`;
